packages/common/src/api/tan-query/useAddToPlaylistFolder.ts

```diff
@@ -0,0 +1,110 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { useDispatch } from 'react-redux'
+
+import { useAppContext } from '~/context/appContext'
+import { FavoriteSource, Name } from '~/models/Analytics'
+import {
+  PlaylistLibrary,
+  PlaylistLibraryID,
+  PlaylistLibraryKind,
+  PlaylistLibraryFolder
+} from '~/models/PlaylistLibrary'
+import { playlistLibraryHelpers } from '~/store/playlist-library'
+import { saveCollection } from '~/store/social/collections/actions'
+import { toast } from '~/store/ui/toast/slice'
+
+import { QUERY_KEYS } from './queryKeys'
+import { useCurrentUserId } from './useCurrentUserId'
+import { usePlaylistLibrary } from './usePlaylistLibrary'
+import { useUpdatePlaylistLibrary } from './useUpdatePlaylistLibrary'
+
+type AddToFolderVariables = {
+  draggingId: PlaylistLibraryID
+  draggingKind: PlaylistLibraryKind
+  folder: PlaylistLibraryFolder
+}
+
+type AddToFolderResult = {
+  updatedLibrary: PlaylistLibrary
+  draggingId: PlaylistLibraryID
+  folder: PlaylistLibraryFolder
+}
+
+const messages = {
+  playlistMovedToFolderToast: (folderName: string) =>
+    `This playlist was already in your library. It has now been moved to ${folderName}!`
+}
+
+/**
+ * Hook to add items to a playlist folder
+ */
+export const useAddToPlaylistFolder = () => {
+  const { data: currentUserId } = useCurrentUserId()
+  const { data: playlistLibrary } = usePlaylistLibrary()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+  const updatePlaylistLibrary = useUpdatePlaylistLibrary()
+  const {
+    analytics: { track, make }
+  } = useAppContext()
+
+  return useMutation<AddToFolderResult, Error, AddToFolderVariables>({
+    mutationFn: async ({ draggingId, folder }: AddToFolderVariables) => {
+      if (!playlistLibrary || !currentUserId) {
+        throw new Error('Missing required data')
+      }
+
+      const updatedLibrary = playlistLibraryHelpers.addPlaylistToFolder(
+        playlistLibrary,
+        draggingId,
+        folder.id
+      )
+
+      await updatePlaylistLibrary.mutateAsync(updatedLibrary)
+
+      return {
+        updatedLibrary,
+        draggingId,
+        folder
+      }
+    },
+    onSuccess: ({ updatedLibrary, draggingId, folder }) => {
+      // Invalidate the playlist library query
+      queryClient.setQueryData(
+        [QUERY_KEYS.playlistLibrary, currentUserId],
+        updatedLibrary
+      )
+
+      // If dragging in a new playlist, save to user collections
+      if (typeof draggingId === 'number') {
```

Comment:
Nit: It feels like a leak to be dealing with drag and drop details inside of a tan-query hook, which is just supposed to be about data fetching and remote data manipulation?
Especially true since this is common and drag and drop isn't a thing on mobile. I'm wondering if there's a way to split out the logic related to manipulating UI items on web and the logic that fetches/updates the playlist_library data.